build.rs

```diff
@@ -102,6 +105,15 @@ pub const LINK_DIRS: &[&str] = &[
 #[cfg(target_os = "linux")]
 pub const SHARED_LIB_DIRS: &[&str] = LINK_DIRS;
 
+#[cfg(target_os = "macos")]
+const MACOS_COMPILER_PATH: &str = "compiler/latest/mac/compiler/lib";
+
+#[cfg(target_os = "macos")]
+pub const LINK_DIRS: &[&str] = &[MACOS_COMPILER_PATH, "mkl/latest/lib"];
+
+#[cfg(target_os = "macos")]
+pub const SHARED_LIB_DIRS: &[&str] = LINK_DIRS;
```

Comment:
I see MACOS_COMPILER_PATH is explicitly skipped in the shared lib dirs check below. Let's just remove it from shared lib dirs so we don't need that explicit skip. SHARED_LIB_DIRS is only used to the path checking, so it isn't actually passed to rustc at all.

```suggestion
#[cfg(target_os = "macos")]
pub const LINK_DIRS: &[&str] = &["compiler/latest/mac/compiler/lib", "mkl/latest/lib"];

#[cfg(target_os = "macos")]
pub const SHARED_LIB_DIRS: &[&str] = &["mkl/latest/lib"];
```